mitodl/ol-django
# Add SoftDeleteModel, ActiveUndeleteManager; tests for same

### What are the relevant tickets?

Came up in https://github.com/mitodl/unified-ecommerce/pull/31/

### Description (What does it do?)

Adds a SoftDeleteModel to the common library, so we can have soft deletions in a common spot. Adds a manager (ActiveUndeleteManager) so models that use the SoftDeleteModel can also optionally default to hiding items that have been deleted.

### How can this be tested?

Automated tests should pass. Your model can inherit from SoftDeleteModel, and that should add an `is_active` flag to the model and override the `delete` function to set the flag to False. Your model can optionally also use the `ActiveUndeleteManager` so that items that are deleted (`is_active == False`) are excluded from the queryset by default.
